config/api_client.go

```diff
@@ -14,6 +14,18 @@ import (
 	"github.com/databricks/databricks-sdk-go/useragent"
 )
 
+// workspaceOrgIdVisitor creates a visitor that adds X-Databricks-Org-Id header
+// with the workspace ID to all requests made by workspace clients
+// It relies on the workspace ID being set if and only if a workspace client is being used.
+func workspaceOrgIdVisitor(cfg *Config) func(r *http.Request) error {
+	return func(r *http.Request) error {
+		if cfg.WorkspaceId != "" {
```

Comment:
suggestion: let's check what happens if a user specifies the wrong workspace ID for a given workspace host.